Commands/CDirectiveFile.cpp

```diff
@@ -562,3 +562,124 @@ void DirectiveObjImport::writeSymData(SymbolData& symData) const
 
 	rel.writeSymbols(symData);
 }
+
+//
+// CDirectiveSymImport
+//
+CDirectiveSymImport::CDirectiveSymImport(const fs::path& fileName)
+{
+	// We may be adding global labels, so start a new section
+	updateSection(++Global.Section);
+
+	this->fileName = getFullPathName(fileName);
+	if (!fs::exists(this->fileName))
+	{
+		Logger::printError(Logger::FatalError,"File %s not found",this->fileName.u8string());
+		return;
+	}
+
+	fs::ifstream file(this->fileName,fs::ifstream::in|fs::ifstream::binary);
+	if (!file.is_open())
+	{
+		Logger::printError(Logger::FatalError,"Could not open file %s",this->fileName.u8string());
+		return;
+	}
+
+	std::string line;
+	size_t l = 0;
+	while (l++, std::getline(file,line,'\n'))
+	{
+		// End line at comment (starting with ;)
+		// \x1A included here as well since No$gba .sym file ends with it
+		size_t lineEnd = line.length();
+		lineEnd = std::min(line.find_first_of(";\r\x1A"),lineEnd);
+		if (lineEnd != std::string::npos && lineEnd > 0)
+		{
+			lineEnd = std::max(line.find_last_not_of(" \t",lineEnd-1)+1,(size_t)0);
+		}
+
+		// Skip empty line
+		if (lineEnd <= 0)
+			continue;
+		line = line.substr(0,lineEnd);
+
+		// Parse address of exactly 8 chars
+		const char* addressStart = line.c_str();
+		char* addressEnd;
+		uint32_t address = strtoul(addressStart,&addressEnd,16);
+		if (addressEnd != addressStart+8)
+		{
+			Logger::printError(Logger::Warning,"Invalid symbol address on line %i of symbols file %s",l,this->fileName);
+			continue;
+		}
+
+		// Skip one or more space or tabs
+		size_t startOfName = line.find_first_not_of(" \t",8);
+		if (startOfName == std::string::npos || startOfName < (8+1))
+		{
+			Logger::printError(Logger::Warning,"Invalid symbol address on line %i of symbols file %s",l,this->fileName);
+			continue;
+		}
+
+		// Rest of the line is the symbol name
+		std::string name = line.substr(startOfName);
+
+		// Create label for this symbol, if it's not a directive and it would be a global label
+		const Identifier identifier = Identifier(name);
```

Comment:
This works for no$gba .sym files (or `-sym` output), but not for files generated by `-sym2` (if `.func` was used, at least). Anything after a comma should also be removed from the identifier.